stratosphere/loader/source/ldr_ecs.cpp

```diff
@@ -28,15 +28,40 @@ namespace ams::ldr::ecs {
             NON_COPYABLE(ExternalContentSource);
             NON_MOVEABLE(ExternalContentSource);
             private:
+                bool has_mounted = false;
                 char device_name[DeviceNameSizeMax];
+                os::ManagedHandle client;
+
+                Result Mount() {
+                    /* Create service. */
+                    Service srv;
+                    serviceCreate(&srv, client.Move());
+                    FsFileSystem fs = { srv };
+                    auto fs_guard = SCOPE_GUARD { fsFsClose(&fs); };
+
+                    /* Try to mount. */
+                    R_UNLESS(fsdevMountDevice(device_name, fs) >= 0, fs::ResultMountNameAlreadyExists());
+                    fs_guard.Cancel();
+
+                    has_mounted = true;
+                    return ResultSuccess();
+                }
             public:
-                ExternalContentSource(const char *dn) {
+                ExternalContentSource(const char *dn, os::ManagedHandle client) : client(std::move(client)) {
                     std::strncpy(this->device_name, dn, sizeof(this->device_name));
                     this->device_name[sizeof(this->device_name) - 1] = '\0';
                 }
 
+                void EnsureMounted() {
+                    if(!has_mounted) {
+                        Mount();
```

Comment:
What happens in the case where Mount() fails?

Reply:
We now silently remove the ECS from the map and pretend it never existed.